File: app/src/main/java/com/azreashade/shadowpurge/UsageAccessUtils.java
package com.azreashade.shadowpurge;

import android.app.AppOpsManager;
import android.content.Context;
import android.content.Intent;
import android.os.Process;
import android.provider.Settings;

public class UsageAccessUtils {
    public static boolean hasUsageAccess(Context c) {
        try {
            AppOpsManager a = (AppOpsManager) c.getSystemService(Context.APP_OPS_SERVICE);
            int mode = a.checkOpNoThrow(AppOpsManager.OPSTR_GET_USAGE_STATS, Process.myUid(), c.getPackageName());
            return mode == AppOpsManager.MODE_ALLOWED;
        } catch (Exception e) {
            return false;
        }
    }

    public static void openUsageAccess(Context c) {
        Intent i = new Intent(Settings.ACTION_USAGE_ACCESS_SETTINGS);
        i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        c.startActivity(i);
    }
}

File: app/src/main/java/com/azreashade/shadowpurge/ui/UsageAccessUtils.kt
package com.azreashade.shadowpurge.ui

import android.app.AppOpsManager
import android.content.Context
import android.os.Process
import android.util.Log

fun checkUsageAccess(context: Context): Boolean {
    try {
        val appOps = context.getSystemService(Context.APP_OPS_SERVICE) as AppOpsManager
        val mode = appOps.checkOpNoThrow(
            AppOpsManager.OPSTR_GET_USAGE_STATS,
            Process.myUid(),
            context.packageName
        )
        return mode == AppOpsManager.MODE_ALLOWED
    } catch (e: Exception) {
        Log.e("UsageAccessUtils", "Error checking usage access", e)
        return false
    }
}
